### PR TITLE
Auto ranging on LCD

### PR DESCRIPTION
**Pull request by trstephen**
I gave up, I need `<string.h>` to do the `float -> string` conversion 😓

### CODE REVIEW

**Comment by trstephen** [comment]
Resolves #9 

(ref not automatically detected by githook?!)

File: src/lcd.c [hunk 1]
```diff
@@ -327,53 +335,51 @@ void LCD_UpdateFreq(float freq){
...
-    for (int i = 0; i < DIGITS_TO_WRITE; i++) {
-        if (orderedDigits[i] == LEADING_ZERO_FLAG) {
-            LCD_SendASCIIChar(" ");
-        } else {
-            LCD_SendInteger(orderedDigits[i]);
-        }
-    }
+    LCD_SendASCIIChar(" "); // overwrite over/under range symbol
+    LCD_SendText(printableVal);
+    LCD_SendASCIIChar(metricVal.prefix);
```
Comment (trstephen) [inline src/lcd.c]: Nit: might as well just use `LCD_SendText` exclusively

File: src/lcd.c [hunk 2]
```diff
@@ -327,53 +335,51 @@ void LCD_UpdateFreq(float freq){
...
-        }
+    // Scale val for metric representation
+    metricVal.value = val / metricVal.multiplier;
+
+    // Determine number of decimal places to display by removing the part of
+    // the number greater than 0 from the total number of available columns.
+    // The remainder will go to the decimal;
+    // TODO: Dynamically determine precision range from LCD_MAX_DIGIT_COLS
+    char* floatPrecisionFormat;
+    if (metricVal.value < 10) {
```
Comment (trstephen) [inline src/lcd.c:368]: It's probably not that much work to add mili units. We can reliably measure frequencies less than 1 Hz so it has a use case.

**Comment by tkun3** [comment]
Looks good!

**Pull request merged**